File: Core/YABACore/src/commonMain/kotlin/dev/subfly/yabacore/model/utils/ThemePreference.kt
package dev.subfly.yabacore.model.utils

enum class ThemePreference {
    LIGHT,
    DARK,
    SYSTEM,
}

fun ThemePreference.uiIconName(): String =
    when (this) {
        ThemePreference.LIGHT -> "sun-03"
        ThemePreference.DARK -> "moon-02"
        ThemePreference.SYSTEM -> "smart-phone-02"
    }

expect fun ThemePreference.uiTitle(): String

File: Darwin/YABACore/Model/Core/ThemePreference.swift
//
//  ThemePreference.swift
//  YABACore
//
//  Parity with Compose `ThemePreference`.
//

import Foundation
import SwiftUI

public enum ThemePreference: String, Sendable, Codable, CaseIterable {
    case light
    case dark
    case system

    public var uiIconName: String {
        switch self {
        case .light: return "sun-03"
        case .dark: return "moon-02"
        case .system: return "smart-phone-02"
        }
    }

    public func getUIIconName() -> String {
        uiIconName
    }

    public func getUITitle() -> LocalizedStringKey {
        switch self {
        case .light: return LocalizedStringKey("Theme Light")
        case .dark: return LocalizedStringKey("Theme Dark")
        case .system: return LocalizedStringKey("Theme System")
        }
    }

    public func getScheme() -> ColorScheme? {
        switch self {
        case .light: return .light
        case .dark: return .dark
        case .system: return nil
        }
    }
}
